ui/source/js/components/game.js

```diff
@@ -103,24 +104,38 @@ class Game extends React.Component {
 				<Stats
 					isBlueTurn={this.state.isBlueTurn}
 					winner={this.state.winner}
-					viewer={this.state.viewer}
 					score={this.state.score}
 					teams={this.state.teams}
 				/>
 				<Board
 					totalPieces={gamePiecesCount}
 					pieces={this.state.pieces}
-					viewer={this.state.viewer}
+					teams={this.state.teams}
+					playerId={this.state.currentPlayerId}
 					winner={this.state.winner}
 					onClick={(i) => this.pieceChosenClick(i)}
 				/>
 				<Controls
-					viewer={this.state.viewer}
 					winner={this.state.winner}
-					onClickSwitchTeams={() => this.switchTeamsClick()}
-					onClickViewMaster={() => this.setViewer(0)}
-					onClickViewBlue={() => this.setViewer(1)}
-					onClickViewRed={() => this.setViewer(2)}
+					teams={this.state.teams}
+					playerId={this.state.currentPlayerId}
+					isBlueTurn={this.state.isBlueTurn}
+					onClickSwitchTeams={() => this.switchTeamTurnClick()}
+					onClickViewMaster={
+						(e) => this.setState({
+							teams: setPlayerIsMaster(this.state.teams, this.state.currentPlayerId, e.target.checked)
+						})
+					}
+					onClickViewBlue={
+						(e) => this.setState({
+							teams: setPlayerTeam(this.state.teams, this.state.currentPlayerId, 1)
+						})
+					}
+					onClickViewRed={
+						(e) => this.setState({
```

Comment:
You've got some repetitive code for the onClickViewMaster, onClickViewBlue, and onClickViewRed callbacks. You could pull this functionality out into a function that each callback could call. That would also reduce the amount of code in your JSX.

Reply:
I think I got this fixed.